lib/api_client.dart

```diff
@@ -51,9 +51,9 @@ class ApiClient {
   // If collectionFormat is 'multi' a key might appear multiple times.
   Future<ApiResponse> invokeAPI(String path, Iterable<QueryParam> queryParams,
       Object body, List<String> authNames, Options options) async {
-    _updateParamsForAuth(authNames, queryParams, options.headers);
+    _updateParamsForAuth(authNames, queryParams as List<QueryParam>, options.headers);
 
-    options.headers.addAll(_defaultHeaderMap);
+    options.headers!.addAll(_defaultHeaderMap);
```

Comment:
This is telling us we are assuming the options.headers to not be null and we probably shouldn't. I would pull that out into a separate field like final options.headers = options.headers || <whatever-type>(). Then its ok to have this line.